Book-A-Desk.Api/BookingsHttpHandler.fs

```diff
@@ -1,37 +1,62 @@
 ﻿namespace Book_A_Desk.Api
 
-open System
-
+open Amazon.DynamoDBv2
+open Microsoft.AspNetCore.Http
 open Giraffe
+open FsToolkit.ErrorHandling
 open FSharp.Control.Tasks.V2.ContextInsensitive
+open System
 
+open Book_A_Desk.Api.Models
 open Book_A_Desk.Domain
+open Book_A_Desk.Domain.CommandHandler
 open Book_A_Desk.Domain.Office.Domain
+open Book_A_Desk.Domain.Reservation
 open Book_A_Desk.Domain.Reservation.Commands
-open Book_A_Desk.Domain.CommandHandler
+open Book_A_Desk.Domain.Reservation.Domain
+open Book_A_Desk.Infrastructure.DynamoDbEventStore
 
-open Book_A_Desk.Api.Models
 type BookingsHttpHandler =
     {
-        HandlePostWith: Booking -> HttpHandler
+        HandlePostWith: Models.Booking -> HttpHandler
     }
 
 module BookingsHttpHandler =
-    let initialize eventStore reservationCommandsFactory =
-
-        let handlePostWith booking = fun next context ->
+    let initialize (provideEventStore : IAmazonDynamoDB -> DynamoDbEventStore) reservationCommandsFactory =
+        let handlePostWith booking = fun next (context : HttpContext) ->
             task {
                 let cmd =
                     {
                         BookADesk.OfficeId = Guid.Parse(booking.Office.Id) |> OfficeId // Consider TryParse and return 400 if not valid
                         Date = booking.Date
                         EmailAddress = EmailAddress booking.User.Email
                     }
-
+ 
+                let eventStore = provideEventStore (context.GetService<IAmazonDynamoDB>())
                 let command = BookADesk cmd
-                let commandHandler = ReservationsCommandHandler.provide eventStore reservationCommandsFactory
+                
+                let handleCommand command = asyncResult {
```

Comment:
Summary of what changed here:
To prevent our Domain becoming async, the Domain has changed a little bit. Instead of the Domain fetching/appending the events, we fetch and append the events in the API. The Domain takes in the previous events and returns the new events to append